nixos/nvidia: add nvidia_dc_565 drivers

edwtjo opened the pull request:
Also remove dc_520 and use nixfmt on module and driver files.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Kiskae reviewed (changes requested):
Let's wait for https://github.com/NixOS/nixpkgs/pull/322537 or move the formatting to a seperate commit. I can't really tell what is formatting and what is an actual change.

Kiskae commented on pkgs/os-specific/linux/nvidia-x11/default.nix:
```diff
@@ -111,6 +115,19 @@ rec {
 
+  dc_565 = generic rec {
+    version = "565.57.01";
+    url = "https://developer.download.nvidia.com/compute/nvidia-driver/redist/nvidia_driver/linux-x86_64/nvidia_driver-linux-x86_64-${version}-archive.tar.xz";
```
Is there a reason you're using the cuda archive driver, rather than the official runfile?

https://docs.nvidia.com/datacenter/tesla/drivers/releases.json lists a driver for `565.57.01`

Kiskae reviewed (changes requested):
Please rebase on `master` and move the formatting to a separate commit.

In addition I'm not sure its a good idea to add even more variations to the builder when a regular runfile is available.

edwtjo commented:
> Please rebase on `master` and move the formatting to a separate commit.

Done

> In addition I'm not sure its a good idea to add even more variations to the builder when a regular runfile is available.

Right and I agree, this was done in haste without much reflection.. I believe I've addressed your comments. This will still need some more testing, which I intend to do next week.

Kiskae commented on pkgs/top-level/linux-kernels.nix:
Definitely don't format this file; it isn't owned by just the nvidia driver.

At best you can use it to ensure the added code is formatted correctly, but leave the other code unchanged,

Kiskae commented on nixos/modules/hardware/video/nvidia.nix:
```diff
@@ -296,7 +296,7 @@ in
             }
             {
-              assertion = cfg.open != null;
+              assertion = cfg.open != null || cfg.datacenter.enable;
```
This will unconditionally cause an error if `hardware.nvidia.datacenter.enable` is `true`. That can't be the intended effect.

Kiskae replied:
Nvm, I read the assertion the wrong way around

Kiskae commented on pkgs/os-specific/linux/nvidia-x11/default.nix:
```diff
@@ -100,20 +100,7 @@ rec {
 
   # data center driver compatible with current default cudaPackages
-  dc = dc_520;
-  dc_520 = generic rec {
```
Your release notes say 'has been deprecated' , but you've removed the driver entirely.

edwtjo committed:
nixos/nvidia: run nixfmt on nvidia related files

edwtjo committed:
nixos/nvidia: add nvidia_dc_565 drivers

Also remove dc_520

edwtjo commented:
All checks have been passed, both maintainers approve on the changes, bundled services tested in datacenter.
